plugins/quota/actor.go

```diff
@@ -0,0 +1,68 @@
+package quota
+
+import (
+	"os"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	gderrors "github.com/gluster/glusterd2/pkg/errors"
+)
+
+type quotadActor struct{}
+
+const (
+	quotaDaemonKey = "enable"
+)
+
+func (actor *quotadActor) Do(v *volume.Volinfo, key string, value string) error {
+	if key != quotaDaemonKey {
+		return nil
+	}
+	quotadDaemon, err := NewQuotad()
+	if err != nil {
+		return err
+	}
+
+	switch value {
+	case "on":
+
+		// Create pidfile dir if not exists
+		err = os.MkdirAll(path.Dir(quotadDaemon.pidfilepath), os.ModeDir|os.ModePerm)
+		if err != nil {
+			return err
+		}
+		// Create logFiledir dir
+		err = os.MkdirAll(path.Dir(quotadDaemon.logfilepath), os.ModeDir|os.ModePerm)
+		if err != nil {
+			return err
+		}
+		err = daemon.Stop(quotadDaemon, true)
+		if err != nil && err != gderrors.ErrProcessNotRunning && err != gderrors.ErrPidFileNotFound {
+			return err
+		}
```

Comment:
Over the long term, we should be sending an RPC to quota daemon to notify about volfile change without needing a restart.